pkg/status/clusteragent/status_apiserver.go

```diff
@@ -29,11 +29,7 @@ func GetDCAStatus(stats map[string]interface{}) {
 	}
 	clusterAgentDetails["Endpoint"] = dcaCl.ClusterAgentAPIEndpoint()
 
-	ver, err := dcaCl.GetVersion()
-	if err != nil {
-		clusterAgentDetails["ConnectionError"] = err.Error()
-		return
-	}
+	ver := dcaCl.Version()
```

Comment:
Question: does this change the current behavior?
With the current code, if there's a rollout of a new version of the DCA, it will get the new version. However, now it will get the cached one if the client has been initialized previously. I haven't checked if it's the case in this code path.

Reply:
If I understand correctly, we refresh the version periodically [here](https://github.com/DataDog/datadog-agent/blob/7a2a2f74db938c0c9e2df5535720b23580d350fa/pkg/util/clusteragent/clusteragent.go#L144) by establishing a new connection every 1 hour (period defined [here](https://github.com/DataDog/datadog-agent/blob/7a2a2f74db938c0c9e2df5535720b23580d350fa/pkg/config/setup/config.go#L444)).

So in practice yes I think the current behaviour is going to change in the sense that in case of a rollout we will need to wait at most 1 hour to get the new version.

Do you think this is an issue? 

If yes, I think what we can do is refresh the version before accessing the cache when running the agent status command.